### DISCUSSION TITLE
Help with securing discord bot token in Replit

### DISCUSSION THREAD

**Discussion by TactualGalaxy** [SudhanPlayz/Discord-MusicBot#1047] [Q&A] [answered]
Hi, I'm an amateur and don't fully understand javascript.

Recently I set up a discord music bot with Replit. It works great, thanks. However, I got a notification from discord that my bot's token has been leaked (Replit makes all of its code public). Replit community advises to use a .env file for a python solution, however it doesn't help with javascript. I don't know what to do.

Please help

**Comment by DarrenOfficial** [comment, marked as answer]
Hello by default we do support the use of `.env` or as repl.it called it `System environment variables.`.

Simply head over to environment variables, see attached image below
![env variables](https://user-images.githubusercontent.com/55406094/185100659-cc1f2e33-1e24-43a9-8907-248392d23bf6.png)

And fill in the key with the name of the "item", and the value as the "value". See attached image
![value](https://user-images.githubusercontent.com/55406094/185101084-621a1210-d0da-43f3-b80f-70255e8f4754.png)

Do this with all the "secrets" you want to hide.
And don't forget to leave this field blank. See attached image
![value env](https://user-images.githubusercontent.com/55406094/185101305-da9d35d3-8c3b-4a22-814c-28c84c1790a7.png)

Do let me know if you have anymore questions. Hope this help!

**Reply by TactualGalaxy** [reply]
Thanks this is very helpful. Thank you for the quick response and screenshot examples, That made it easier to understand. Thank you :smiley:
